tests/test_cli.py

```diff
@@ -440,3 +440,12 @@ def test_no_candidates_pre():
 
         assert out.exit_code == 2
         assert 'Tried pre-versions:' in out.output
+
+
+@pytest.mark.usefixtures('pip_conf')
+def test_default_index_url():
+    status, output = _invoke(['pip-compile', '--help'])
+    output = output.decode('utf-8')
+
+    assert status == 0
+    assert 'http://example.com' in output
```

Comment:
I think this should be more specific, and could probably also be asserted from a test that runs `--help` again (less overhead).

Reply:
Re "more specific": it should match other parts of the line printing this, so that it would not match e.g. a copyright line etc.. ;)

Reply:
> Re "more specific": it should match other parts of the line printing this, so that it would not match e.g. a copyright line etc.. ;)

Good point! Though i'm not sure how to test properly multiline string in output. See:

```
         -i, --index-url TEXT            Change index URL (defaults to
                                         http://example.com)
```

Do you have any suggestions?

> also be asserted from a test that runs `--help` again (less overhead).

I'd rather to make a separate test for that. In ideal world i'd prefer one assert per one test :)

Reply:
What about this one? Looks a bit hacky though.
```python
assert "Change index URL (defaults to http://example.com)" in " ".join(output.split(" "))
```

Reply:
No, it doesn't work.

Reply:
Anyways, i've tested a whole string.  Check it out please.

Reply:
@blueyed finally managed to fix failing test on win+py27. Reported an issue  https://github.com/pallets/click/issues/1231.